server/utils/premises/index.ts

```diff
@@ -161,7 +171,74 @@ export const premisesTableRows = (premisesSummaries: Array<Cas1PremisesBasicSumm
         textValue(p.name),
         textValue(p.apCode),
         textValue(p.bedCount.toString()),
-        htmlValue(linkTo(paths.premises.show, { premisesId: p.id }, { text: 'View', hiddenText: `about ${p.name}` })),
+        htmlValue(
+          linkTo(managePaths.premises.show, { premisesId: p.id }, { text: 'View', hiddenText: `about ${p.name}` }),
+        ),
       ]
     })
 }
+
+type TabKey = 'upcoming' | 'current' | 'historic'
+
+export const TAB_TEXT_MAP: Record<TabKey, string> = {
+  upcoming: 'Upcoming',
+  current: 'Current',
+  historic: 'Historical',
+}
+
+export const premisesTabItems = (premises: Cas1PremisesSummary, activeTab?: string): Array<TabItem> => {
+  const getSelfLink = (tab: string): string =>
+    `${managePaths.premises.show({ premisesId: premises.id })}${createQueryString(
+      {
+        activeTab: tab,
+      },
+      { addQueryPrefix: true },
+    )}`
+  return Object.entries(TAB_TEXT_MAP).map(([key, label]) => {
+    return { text: label, active: activeTab === key, href: getSelfLink(key) }
+  })
+}
+
+type ColumnDefinition = {
+  title: string
+  fieldName: Cas1SpaceBookingSummarySortField
+}
+const baseColumns: Array<ColumnDefinition> = [
+  { title: 'Name and CRN', fieldName: 'personName' },
+  { title: 'Tier', fieldName: 'tier' },
+  { title: 'Arrival date', fieldName: 'canonicalArrivalDate' },
+  { title: 'Departure date', fieldName: 'canonicalDepartureDate' },
+]
+const keyWorkerColumn: ColumnDefinition = { title: 'Key worker', fieldName: 'keyWorkerName' }
+
+const columnMap: Record<TabKey, Array<ColumnDefinition>> = {
+  upcoming: [...baseColumns, keyWorkerColumn],
+  current: [...baseColumns, keyWorkerColumn],
+  historic: baseColumns,
+}
+
+export const placementTableHeader = (
+  activeTab: string,
+  sortBy: Cas1SpaceBookingSummarySortField,
+  sortDirection: SortDirection,
+  hrefPrefix: string,
+): Array<TableCell> => {
+  return columnMap[activeTab].map(({ title, fieldName }: ColumnDefinition) =>
+    sortHeader<Cas1SpaceBookingSummarySortField>(title, fieldName, sortBy, sortDirection, hrefPrefix),
+  )
+}
+
+export const placementTableRows = (
+  activeTab: string,
+  premisesId: string,
+  placements: Array<Cas1SpaceBookingSummary>,
+): Array<TableRow> =>
+  placements.map(({ id, person, tier, canonicalArrivalDate, canonicalDepartureDate, keyWorkerAllocation }) => [
+    htmlValue(
+      `<a href="${managePaths.premises.placements.show({ premisesId, bookingId: id })}" data-cy-id="${id}">${(person as PersonWithName).name}, ${person.crn}</a>`,
```

Comment:
This should probably use one of the available person name utilities, though not sure which exactly -- possibly `laoName`? https://github.com/ministryofjustice/hmpps-approved-premises-ui/blob/main/server/utils/personUtils.ts#L24